src/aggregated/account_details.rs

```diff
@@ -0,0 +1,64 @@
+use actix::Addr;
+
+use near_client::{Query, ViewClientActor};
+use near_indexer::near_primitives::hash::CryptoHash;
+use near_indexer::near_primitives::types::{Balance, BlockId, BlockReference};
+use near_indexer::near_primitives::views::{AccountView, QueryRequest, QueryResponseKind};
+
+pub(crate) async fn get_account_balance(
+    view_client: &Addr<ViewClientActor>,
+    account_id: &str,
+    block_height: u64,
+) -> Result<Balance, String> {
+    get_account_view_for_block_height(view_client, account_id, block_height)
+        .await
+        .map(|acc| acc.amount)
+}
+
+pub(crate) async fn get_contract_code_hash(
+    view_client: &Addr<ViewClientActor>,
+    account_id: &str,
+    block_height: u64,
+) -> Result<CryptoHash, String> {
+    get_account_view_for_block_height(view_client, account_id, block_height)
+        .await
+        .map(|acc| acc.code_hash)
+}
+
+async fn get_account_view_for_block_height(
+    view_client: &Addr<ViewClientActor>,
+    account_id: &str,
+    block_height: u64,
+) -> Result<AccountView, String> {
+    let block_reference = BlockReference::BlockId(BlockId::Height(block_height));
+    let request = QueryRequest::ViewAccount {
+        account_id: account_id
+            .parse()
+            .map_err(|_| "Failed to parse `account_id`")?,
```

Comment:
I suggest this function requires `account_id` to be `AccountId`, and all the functions above should also require `AccountId`, and validate the account id when we read the value (e.g. when it is a foundation account, we can use `"foundation.near".parse().expect()` since those are obviously valid).